File: backend/app/memory/api/schemas/discoveries.py
from typing import Optional
from pydantic import BaseModel, Field


class PatternExcerptResponse(BaseModel):
    entryId: str
    quote: str
    entryDate: Optional[str] = None


class PatternCardResponse(BaseModel):
    id: str
    description: str
    evidence: str
    excerpts: list[PatternExcerptResponse] = Field(default_factory=list)


class JourneyStateResponse(BaseModel):
    status: str = Field(..., description="'empty' (before model output) or 'active' (after model output)")
    totalEntries: int = 0
    totalWords: int = 0
    firstEntryDate: Optional[str] = None
    lastEntryDate: Optional[str] = None
    lastModelUpdate: Optional[str] = None
    modelVersion: int = 0


class MomentItemResponse(BaseModel):
    id: str
    type: str = Field(..., description="Moment type: 'first_entry', 'baseline_ready', 'model_update'")
    date: str
    title: str
    description: str


class DiscoveriesResponse(BaseModel):
    status: str = "ok"
    journey: JourneyStateResponse
    patterns: list[PatternCardResponse] = Field(default_factory=list)
    activeThemes: list[str] = Field(default_factory=list)
    moments: list[MomentItemResponse] = Field(default_factory=list)

File: frontend/src/types/discoveries.ts
export interface PatternExcerpt {
  entryId: string;
  quote: string;
  entryDate?: string | null;
}

export interface PatternCard {
  id: string;
  description: string;
  evidence: string;
  excerpts: PatternExcerpt[];
}

export interface JourneyState {
  status: 'empty' | 'active';
  totalEntries: number;
  totalWords: number;
  firstEntryDate?: string | null;
  lastEntryDate?: string | null;
  lastModelUpdate?: string | null;
  modelVersion: number;
}

export interface MomentItem {
  id: string;
  type: string;
  date: string;
  title: string;
  description: string;
}

export interface DiscoveriesPayload {
  status: string;
  journey: JourneyState;
  patterns: PatternCard[];
  activeThemes: string[];
  moments: MomentItem[];
}
